dash/include/dash/algorithm/Sort.h

```diff
@@ -0,0 +1,1177 @@
+#ifndef DASH__ALGORITHM__SORT_H__
+#define DASH__ALGORITHM__SORT_H__
+
+#include <algorithm>
+#include <functional>
+#include <iterator>
+#include <type_traits>
+#include <vector>
+
+#include <dash/Array.h>
+#include <dash/Atomic.h>
+#include <dash/Exception.h>
+#include <dash/GlobPtr.h>
+#include <dash/dart/if/dart.h>
+
+#include <dash/algorithm/Copy.h>
+#include <dash/algorithm/Fill.h>
+#include <dash/algorithm/LocalRange.h>
+#include <dash/algorithm/MinMax.h>
+#include <dash/algorithm/Transform.h>
+
+namespace dash {
+
+#ifdef DOXYGEN
+
+/**
+ * Sorts the elements in the range, defined by \c [begin, end) in ascending
+ * order. The order of equal elements is not guaranteed to be preserved.
+ *
+ * Elements are sorted using operator<.
+ *
+ * In terms of data distribution, source and destination ranges passed to
+ * \c dash::sort must be global (\c GlobIter<ValueType>).
+ *
+ * The operation is collective among the team of the owning dash container.
+ *
+ * Example:
+ *
+ * \code
+ *       dash::Array<int> arr(100);
+ *       dash::generate(arr.begin(), arr.end());
+ *       dash::sort(array.begin(),
+ *                  array.end();
+ * \endcode
+ *
+ * \ingroup  DashAlgorithms
+ */
+template <class GlobRandomIt>
+void sort(GlobRandomIt begin, GlobRandomIt end);
```

Comment:
Clarify documentation: this call is only suitable for arithmetic elements while complex types require the use of a hash function returning an arithmetic type (having `operator<` defined on the type is not sufficient)